lib/modules/manager/pipenv/artifacts.ts

```diff
@@ -289,16 +139,13 @@ export async function updateArtifacts({
       await deleteLocalFile(lockFileName);
     }
     const cmd = 'pipenv lock';
-    const tagConstraint = await getPythonConstraint(
-      pipfileName,
-      newPipfileContent,
-      existingLockFileContent,
-      config,
-    );
-    const pipenvConstraint = getPipenvConstraint(
-      existingLockFileContent,
-      config,
-    );
+    const pipfileDir = getParentDir(ensureLocalPath(pipfileName));
+    const tagConstraint =
+      config?.constraints?.python ??
```

Comment:
```suggestion
      config.constraints?.python ??
```
config is never null or undefined here